### ISSUE TITLE
False positive extension_access_modifier with Xcode 11.4

### ISSUE THREAD

**Issue by CraigSiemens** [realm/SwiftLint#3150]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

`extension_access_modifier` reports a violation when an extension contains a method with an explicit access modifier and a method with an implicit modifier of `internal`.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint 
Loading configuration from '.swiftlint.yml'
Linting Swift files at paths 
Linting 'Test.swift' (1/1)
/Users/craig/Desktop/test/Test.swift:3:1: warning: Extension Access Modifier Violation: Prefer to use extension access modifiers (extension_access_modifier)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Environment

* SwiftLint version: 0.39.1
* Installation method used: Homebrew
* Paste your configuration file:

```yml
opt_in_rules:
  - extension_access_modifier
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? No
  If so, paste their relative paths and respective contents.
* Which Xcode version are you using (check `xcodebuild -version`)? 11.4
* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
struct Foo {}

// Extension Access Modifier Violation: Prefer to use extension access modifiers (extension_access_modifier)
extension Foo {
    func setup() {}
    public func update() {}
}
```

**Comment by marcelofabri** [comment]
Fixed in https://github.com/realm/SwiftLint/pull/3153
